python-threatexchange/threatexchange/cli/experimental_fetch.py

```diff
@@ -36,75 +44,118 @@ def init_argparse(cls, ap) -> None:
+            self.deleted_count = 0
+            self.indicator_store = dataset.get_indicator_store(privacy_group)
+            dataset.load_indicator_cache(self.indicator_store)
+            self.stop_time = request_time if self.stop_time is None else self.stop_time
+            next_page = None
+            if self.continuation:
+                checkpoint = dataset.get_indicator_checkpoint(privacy_group)
+                self.start_time = checkpoint["last_stop_time"]
+                self.stop_time = request_time
+                self.threat_types = checkpoint["threat_types"]
+                next_page = checkpoint["url"]
+                if (
+                    request_time - checkpoint["last_run_time"]
+                    > self.DEFAULT_REFETCH_SEC
+                ):
+                    print("It's been a long time since a full fetch, forcing one now.")
+                    self.start_time = 0
+                    next_page = None
 
-        while more_to_fetch:
-            result = TE.Net.getThreatUpdates(
-                dataset.config.privacy_groups[0],
-                start_time=self.start_time,
-                stop_time=self.stop_time,
-                owner=self.owner,
-                threat_type=self.threat_types,
-                additional_tags=self.additional_tags,
-                next_page=next_page,
-            )
-            if "data" in result:
-                self._process_indicators(result["data"])
-            more_to_fetch = "paging" in result and "next" in result["paging"]
-            next_page = result["paging"]["next"] if more_to_fetch else None
+            more_to_fetch = True
+            remaining_attempts = 5
+            while more_to_fetch:
+                try:
+                    result = TE.Net.getThreatUpdates(
+                        privacy_group,
+                        next_page,
+                        start_time=self.start_time,
+                        stop_time=self.stop_time,
+                        threat_types=self.threat_types,
+                        limit=self.limit,
+                    )
+                    if "data" in result:
+                        self._process_indicators(result["data"])
+                    more_to_fetch = "paging" in result and "next" in result["paging"]
+                    next_page = result["paging"]["next"] if more_to_fetch else ""
+                except Exception as e:
+                    remaining_attempts -= 1
+                    print(f"The following error occured:\n{e}")
+                    if type(e) is urllib.error.HTTPError:
+                        print(e.read())
+                    if remaining_attempts > 0:
+                        print(f"\nTrying again {remaining_attempts} more times.")
+                        time.sleep(5)
+                        continue
+                    else:
+                        print(
+                            "\n5 consecutive errors occured, saving state and shutting down!"
```

Comment:
(blocking): Move this into a constant so if we change the number we don't have to hunt through the code to find all the print statements to update